Documentation for HttpData (#987)
zio/zio-http: docs/website/docs/v1.x/dsl/http-data/index.md

# HttpData
`HttpData` is a domain to model content for `Requests` and `Response`. ZIO HTTP uses Netty at it's core and Netty handles content as `ByteBuf`. `HttpData` helps you decode and encode this content into simpler, easier to use data types while creating a Request or Response.
## Server-side usage of `HttpData`
On the server-side, `ZIO-HTTP` is adding `HttpData.Empty` as the pre-defined data to the `Response`. You may replace it with another `HttpData`.

To attach data to a response you can use the `Response` constructors.
```scala
  val res: Response = Response( data = HttpData.fromString("Some String"))
```
## Client-side usage of `HttpData`

> On the client-side, `ZIO-HTTP` is adding `HttpData.Empty` as the pre-defined data to the `Request`. You may replace it with another `HttpData`.

Reviewer: we can mention, for testing purposes we can use.

To create an `HttpData` that encodes a ByteBuf you can use `HttpData.fromByteBuf`.
```scala
  val binaryByteBufHttpData: HttpData = HttpData.fromByteBuf(Unpooled.copiedBuffer("Some string", CharsetUtil.UTF_8))
```
### Creating an HttpData from `Chunk of Bytes`
To create an `HttpData` that encodes chunk of bytes you can use `HttpData.fromChunk`.
```scala
  val chunkHttpData: HttpData = HttpData.fromChunk(Chunk.fromArray("Some Sting".getBytes(CharsetUtil.UTF_8)))
```
### Creating an HttpData from a `Stream`
To create an `HttpData` that encodes a Stream you can use `HttpData.fromStream`.
- Using a Stream of Bytes
```scala
  val streamHttpData: HttpData = HttpData.fromStream(ZStream.fromChunk(Chunk.fromArray("Some String".getBytes(HTTP_CHARSET))))
```
- Using a Stream of String
```scala
  val streamHttpData: HttpData = HttpData.fromStream(ZStream("a", "b", "c"), CharsetUtil.UTF_8)
```
### Creating an HttpData from a `File`
To create an `HttpData` that encodes a File you can use `HttpData.fromFile`.
```scala
  val fileHttpData: HttpData = HttpData.fromFile(new io.File(getClass.getResource("/fileName.txt").getPath))
```
## Converting `HttpData` to `ByteBuf`

> To convert an `HttpData` to `ButrBuf`  you can call the `toButeBuf` method on it, which returns a `Task[ByteBuf]`.

Reviewer: ```suggestion
To convert an `HttpData` to `ByteBuf`  you can call the `toButeBuf` method on it, which returns a `Task[ByteBuf]`.
```

# HttpData
`HttpData` is a domain to model content for `Request`, `Response` and `ClientRequest`. ZIO HTTP uses Netty at it's core and Netty handles content as `ByteBuf`. `HttpData` helps you decode and encode this content into simpler, easier to use data types while creating a Request or Response.
## Server-side usage of `HttpData`
On the server-side, `ZIO-HTTP` models content in `Request` and `Response` as `HttpData` with `HttpData.Empty` as the default value.

> To attach content while creating a `Response` you can use the `Response` constructor.

Reviewer 2: ```suggestion
To add content while creating a `Response` you can use the `Response` constructor.
```

# HttpData

> `HttpData` is a domain to model the content that is to be written in HttpChannel. `HttpRequest` and `HttpResponse` store content as `ByteBuf`. `HttpData` helps you decode and encode this content into simpler, easier to use data types while creating a Request or Response for ZIO HTTP.

Reviewer: ```suggestion
`HttpData` is a domain to model the content that is to be written on the Netty's HttpChannel.
```

## Server-side usage of `HttpData`
On the server-side, `ZIO-HTTP` is adding `HttpData.Empty` as the pre-defined data to the `Response`. You may replace it with another `HttpData`.

> To attach data to a response you can use the `Response` constructors.

Reviewer: ```suggestion
To add data to a response you can use the `Response` constructors that takes `HttpData.Empty` as the default value
```

```scala
  val res: Response = Response( data = HttpData.fromString("Some String"))
```
## Client-side usage of `HttpData`
On the client-side, `ZIO-HTTP` is adding `HttpData.Empty` as the pre-defined data to the `Request`. You may replace it with another `HttpData`.

> To attach data to a request you can use the `Response` constructors.

Reviewer: Same as above.

```scala
  val req: Request = Request(data = HttpData.fromString("Some String"))
```
## Creating an HttpData

> `HttpData` can be created using a lot of constructors.

Reviewer: This line can be removed.

> ### Creating an empty HttpData

Reviewer: We can remove this para
